includes/class-wc-background-swedbank-pay-queue.php

```diff
@@ -0,0 +1,249 @@
+
+if ( ! class_exists( 'WC_Background_Process', false ) ) {
+	include_once WC_ABSPATH . '/includes/abstracts/class-wc-background-process.php';
+}
+
+/**
+ * Class WC_Background_Swedbank_Queue
+ */
+class WC_Background_Swedbank_Pay_Queue extends WC_Background_Process {
+	/**
+	 * @var WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * Initiate new background process.
+	 */
+	public function __construct() {
+		$this->logger = wc_get_logger();
+
+		// Uses unique prefix per blog so each blog has separate queue.
+		$this->prefix = 'wp_' . get_current_blog_id();
+		$this->action = 'wc_swedbank_pay_queue';
+
+		// Dispatch queue after shutdown.
+		add_action( 'shutdown', array( $this, 'dispatch_queue' ), 100 );
+
+		parent::__construct();
+	}
+
+	/**
+	 * Schedule fallback event.
+	 */
+	protected function schedule_event() {
+		if ( ! wp_next_scheduled( $this->cron_hook_identifier ) ) {
+			wp_schedule_event(
+				time() + MINUTE_IN_SECONDS,
+				$this->cron_interval_identifier,
+				$this->cron_hook_identifier
+			);
+		}
+	}
+
+	/**
+	 * Get batch.
+	 *
+	 * @return \stdClass Return the first batch from the queue.
+	 */
+	protected function get_batch() {
+		global $wpdb;
+
+		$table        = $wpdb->options;
+		$column       = 'option_name';
+		$key_column   = 'option_id';
+		$value_column = 'option_value';
+
+		if ( is_multisite() ) {
+			$table        = $wpdb->sitemeta;
+			$column       = 'meta_key';
+			$key_column   = 'meta_id';
+			$value_column = 'meta_value';
+		}
+
+		$key = $wpdb->esc_like( $this->identifier . '_batch_' ) . '%';
+
+		$results = array();
+
+		// phpcs:disable
+		$data    = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT * FROM {$table} WHERE {$column} LIKE %s ORDER BY {$key_column} ASC",
+			$key ) ); // @codingStandardsIgnoreLine.
+		// phpcs:enable
+
+		foreach ( $data as $id => $result ) {
+			$task = array_filter( (array) maybe_unserialize( $result->$value_column ) );
+
+			$batch       = new \stdClass();
+			$batch->key  = $result->$column;
+			$batch->data = $task;
+
+			$results[ $id ] = $batch;
+
+			// Create Sorting Flow by Transaction Number
+			$sorting_flow[ $id ] = 0;
+			$webhook             = json_decode( $task[0]['webhook_data'], true );
+			if ( JSON_ERROR_NONE !== json_last_error() ) {
+				if ( $webhook && isset( $webhook['transaction']['number'] ) ) {
+					$sorting_flow[ $id ] = $webhook['transaction']['number'];
+				}
+			}
+		}
+
+		// Sorting
+		array_multisort( $sorting_flow, SORT_ASC, SORT_NUMERIC, $results );
+		unset( $data, $sorting_flow );
+
+		// Get first result
+		$batch = array_shift( $results );
+
+		return $batch;
+	}
+
+	/**
+	 * Log message.
+	 *
+	 * @param $message
+	 */
+	private function log( $message ) {
+		$this->logger->info( $message, array( 'source' => 'wc_swedbank_pay_queue' ) );
+	}
+
+	/**
+	 * Code to execute for each item in the queue.
+	 *
+	 * @param mixed $item Queue item to iterate over.
+	 *
+	 * @return bool
+	 */
+	protected function task( $item ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method task() has an NPath complexity of 1445. The configured NPath complexity threshold is 200.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-checkout/pullRequest?prid=5510424)